lasagne/layers/input.py

```diff
@@ -67,6 +67,9 @@ def __init__(self, shape, input_var=None, name=None, **kwargs):
             if input_var.ndim != ndim:
                 raise ValueError("shape has %d dimensions, but variable has "
                                  "%d" % (ndim, input_var.ndim))
+            broad_pattern = [s == 1 for s in self.shape]
+            input_var = T.patternbroadcast(input_var, broad_pattern)
```

Comment:
I thought about that as well, but am not sure if this could introduce problems. If you do:
```python
input_var = T.matrix()
net = InputLayer((100, 1), input_var)
```
then `net.input_var != input_var`. Users might rely on being able to find the input layer for a particular input variable by such a direct comparison. That's why I was leaning towards including this in `lasagne.layers.get_output()`, if at all. I'm unsure.